### PR TITLE
vte, gnome-console: add darwin support

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

As in the title.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 183352` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>appvm</li>
    <li>mate.caja-extensions</li>
    <li>mate.mate-terminal</li>
  </ul>
</details>
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>germinal</li>
    <li>gtkd</li>
    <li>kermit-terminal</li>
    <li>termite (termite-unwrapped)</li>
    <li>vte</li>
  </ul>
</details>

File: pkgs/development/libraries/vte/default.nix [hunk 1]
```diff
@@ -78,6 +78,8 @@ stdenv.mkDerivation rec {
 
   mesonFlags = lib.optionals (!systemdSupport) [
     "-D_systemd=false"
+  ] ++ lib.optionals stdenv.isDarwin [
+    "-D_b_symbolic_functions=false"
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/vte/default.nix:82]: Can you add a comment why this is required?

Comment (wegank) [reply pkgs/development/libraries/vte/default.nix:83]: Done.

File: pkgs/applications/terminal-emulators/gnome-console/default.nix [hunk 2]
```diff
@@ -51,6 +52,11 @@ stdenv.mkDerivation rec {
     wrapGAppsHook
   ];
 
+  mesonFlags = lib.optionals (!stdenv.isLinux) [
+    "-Dnautilus=disabled"
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/gnome-console/default.nix:56]: Werror should always be passed

Comment (wegank) [reply pkgs/applications/terminal-emulators/gnome-console/default.nix:65]: Fetched [upstream patch](https://gitlab.gnome.org/GNOME/console/-/commit/0e29a417d52e27da62f5cac461400be6a764dc65).

**Pull request merged**